include/seqan3/core/concepts/core_detail.hpp

```diff
@@ -0,0 +1,313 @@
+static_assert(!seqan3::copy_constructible_concept<seqan3::detail::test_core_concepts::type_b>);
+
+// Check movable_concept
+static_assert(seqan3::movable_concept<seqan3::detail::test_core_concepts::type_b>);
+static_assert(!seqan3::movable_concept<seqan3::detail::test_core_concepts::type_d>);
+
+// Check copyable_concept
+static_assert(seqan3::copyable_concept<seqan3::detail::test_core_concepts::type_a>);
+static_assert(!seqan3::copyable_concept<seqan3::detail::test_core_concepts::type_b>);
+
+// Check assignable_concept
+static_assert(seqan3::assignable_concept<seqan3::detail::test_core_concepts::type_a&,
+                                         seqan3::detail::test_core_concepts::type_a const &>);
+static_assert(seqan3::assignable_concept<seqan3::detail::test_core_concepts::type_c&,
+                                         seqan3::detail::test_core_concepts::type_b const &>);
+static_assert(!seqan3::assignable_concept<seqan3::detail::test_core_concepts::type_a&,
+                                          seqan3::detail::test_core_concepts::type_c&>);
+
+// Check swappable_concept
+static_assert(seqan3::swappable_concept<seqan3::detail::test_core_concepts::type_a&,
+                                        seqan3::detail::test_core_concepts::type_a&>);
+static_assert(!seqan3::swappable_concept<seqan3::detail::test_core_concepts::type_b,
+                                         seqan3::detail::test_core_concepts::type_c>);
+
+// Check semi_regular_concept
+static_assert(seqan3::semi_regular_concept<seqan3::detail::test_core_concepts::type_a>);
+static_assert(seqan3::semi_regular_concept<seqan3::detail::test_core_concepts::type_c>);
+static_assert(!seqan3::semi_regular_concept<seqan3::detail::test_core_concepts::type_b>);
+static_assert(!seqan3::semi_regular_concept<seqan3::detail::test_core_concepts::type_d>);
+
+// Check regular_concept
+static_assert(!seqan3::regular_concept<seqan3::detail::test_core_concepts::type_a>);
+static_assert(!seqan3::regular_concept<seqan3::detail::test_core_concepts::type_b>);
+static_assert(seqan3::regular_concept<seqan3::detail::test_core_concepts::type_c>);
+static_assert(!seqan3::regular_concept<seqan3::detail::test_core_concepts::type_d>);
+
+// Check invocable_concept
+static_assert(!seqan3::invocable_concept<seqan3::detail::test_core_concepts::type_a, int, double,
+                                         seqan3::detail::test_core_concepts::type_b>);
+static_assert(seqan3::invocable_concept<std::random_device>);
+static_assert(seqan3::invocable_concept<seqan3::detail::test_core_concepts::type_c, int, double,
+                                        seqan3::detail::test_core_concepts::type_b>);
+
+// Check regular_invocable_concept
+static_assert(!seqan3::regular_invocable_concept<seqan3::detail::test_core_concepts::type_a, int, double,
+                                                 seqan3::detail::test_core_concepts::type_b>);
+//TODO(rrahn): Should not meet the regular_invocable_concept
```

Comment:
Well yes, it needs to be investigated, but it is right now not so important. But a ```random_device``` should not be of ```regular_invocable_concept```, because it requires the equality preserving entity between runs of the same function with the same input.